python/requirements.in

```diff
@@ -0,0 +1,4 @@
+astropy==6.0.0
+fastapi==0.105.0
+mangum==0.17.0
+requests==2.31.0
```

Comment:
Explain please.

Reply:
For Python (JavaScript), the requirements.in (package.json) file defines direct dependencies of the project and constrains the range of compatible versions of those dependencies. The requirements.txt (package-lock.json) file, which is automatically generated from the former file, captures a particular dependency solution with concrete version numbers. Generally, the requirements.in file should express ranges of versions but not literal version numbers.

For example, we are probably going to require a version of Astropy that is API compatible with 6.0.0 because of my patches that went into that release. According to [SemVer](https://semver.org) rules, a compatible version would be greater than or equal to 6.0.0 but less than 7.0.0. So in requirements.in, we might express that by writing `astropy~=6.0.0` or (roughly equivalently) `astropy>=6.0.0,<7`.

Reply:
FastAPI suggests that you pin the version (https://fastapi.tiangolo.com/deployment/versions/) for deployment.

Reply:
Well, yes, and we do --- in requirements.txt. But don't pin it in requirements.in.

If and when you find that we are depending on a specific feature of FastAPI that was added in version X.Y.Z, then you can update requirements.in to say: `fastapi>=X.Y.Z`.

Reply:
How often do you envisage that we'll run pip-compile then?

Reply:
We'll run pip-compile _by hand_ pretty infrequently --- only when we are adding dependencies.

However, every week Dependabot will scan our repository's dependencies and open PRs to update any of our dependencies that have new releases. We'll be able to review those PRs and either merge them (if the risk is low and any tests pass) or reject them (if the risk of updating is high or the tests fail).

Reply:
You can see what those Dependabot PRs look like here: https://github.com/nasa-gcn/gcn.nasa.gov/pulls?q=is%3Apr+author%3Aapp%2Fdependabot

Reply:
OK I guess I misunderstood the interplay between requirements.in and Dependabot. I assume it'd be updating the values in that file, then would generate requirements.txt from it. 

So anyway, I unpinned in this commit c177e84.